src/category_theory/limits/shapes/normal_mono.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison, Bhavik Mehta
+-/
+import category_theory.limits.shapes.regular_mono
+import category_theory.limits.shapes.kernels
+
+/-!
+# Definitions and basic properties of normal monomorphisms and epimorphisms.
+
+A normal monomorphism is a morphism that is the kernel of some other morphism.
+
+We give the construction `normal_mono → regular_mono` as well as the dual construction for normal
+epimorphisms. We show equivalences reflect normal monomorphisms, and that the pullback of a normal
+monomorphism is normal.
```

Comment:
```suggestion
We give the construction `normal_mono → regular_mono` (`category_theory.normal_mono.regular_mono`)
as well as the dual construction for normal epimorphisms.
We show equivalences reflect normal monomorphisms, and that the pullback of a normal
monomorphism is normal.
```
And maybe some other lemma links would be good too. Note that without `category_theory.`, the links don't work.

Reply:
Is this the expected style in mathlib now? I wasn't aware of this - it's not mentioned anywhere in https://leanprover-community.github.io/contribute/doc.html#examples as far as I can tell.

Reply:
I don't actually know, it just seems like it would be good if I didn't have to look for the lemma the docs are describing to me.

Reply:
alright, amended